File: 0739-daily-temperatures/0739-daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temp: List[int]) -> List[int]:
        ans = [0]*len(temp)
        stk = [0]
        for i in range(len(temp)):
            while stk and temp[i] > temp[stk[-1]]:
                tempo = stk.pop()
                ans[tempo] = i - tempo
            stk.append(i)
        return ans
            

File: 0739-daily-temperatures/0739-daily-temperatures.js
/**
 * @param {number[]} temperatures
 * @return {number[]}
 */
var dailyTemperatures = function(temperatures) {
    const stack = [];
    const ans = new Array(temperatures.length).fill(0);

    for(let i = 0; i < temperatures.length; i++){
        while(stack.length > 0 && temperatures[stack[stack.length - 1]] < temperatures[i]){
            day = stack.pop();
            ans[day] = i - day;

        }
        stack.push(i);
    }

    return ans;
    
};